File: 2573-remove-nodes-from-linked-list/remove-nodes-from-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNodes(ListNode head) {
        if(head == null) return null;

        ListNode node = head;
        ListNode greaternode = removeNodes(node.next);

        node.next = greaternode;
        if(greaternode == null || node.val >= greaternode.val){
            return node;
        }
        return greaternode;
    }
}

File: 2573-remove-nodes-from-linked-list/remove-nodes-from-linked-list.js
var removeNodes = function(head) {
    let prev = null;
    let curr = head;
    while (curr !== null) {
        let next = curr.next;
        curr.next = prev;
        prev = curr;
        curr = next;
    }

    let dummyHead = new ListNode(-1);
    let tempPrev = dummyHead;
    curr = prev;

    while (curr !== null) {
        if (curr.val >= tempPrev.val) {
            tempPrev.next = curr;
            tempPrev = curr;
            curr = curr.next;
        } else {
            curr = curr.next;
        }
    }
    tempPrev.next = null;

    let newPrev = null;
    let newCurr = dummyHead.next;
    while (newCurr !== null) {
        let next = newCurr.next;
        newCurr.next = newPrev;
        newPrev = newCurr;
        newCurr = next;
    }

    return newPrev;
};